proxy.go

```diff
@@ -0,0 +1,32 @@
+package proxy
+
+import (
+	"fmt"
+	"net"
+	"net/http"
+	"strings"
+	"time"
+
+	"github.com/getlantern/golog"
+	"github.com/getlantern/ops"
+)
+
+var (
+	log = golog.LoggerFor("proxy")
+)
+
+type DialFunc func(network, addr string) (conn net.Conn, err error)
+
+type Interceptor func(op ops.Op, w http.ResponseWriter, req *http.Request)
```

Comment:
Yes, now that we're not pipelining anymore that makes sense, as there'll always be just one error.